projects/driver_controls/src/gpio_expander.c

```diff
@@ -24,39 +35,70 @@ static StatusCode prv_pin_is_valid(GPIOExpanderPin pin) {
   return STATUS_CODE_OK;
 }
 
+// Set a specific bit in a given register
+static void prv_set_bit(uint8_t reg, GPIOExpanderPin pin, bool bit) {
+  uint8_t data;
+  i2c_read_reg(s_i2c_port, MCP23008_ADDRESS, reg, &data, 1);
+  if (bit) {
+    data |= (1 << pin);
+  } else {
+    data &= ~(1 << pin);
+  }
+  i2c_write_reg(s_i2c_port, MCP23008_ADDRESS, reg, &data, 1);
+}
+
+static void prv_debounce_delay(SoftTimerID timer_id, void *context) {
```

Comment:
Any reason for not using the same debounce module here?

Reply:
The debounce module works only for the microcontroller pins. The gpio expander deboucer is also responsible for calling the input callbacks, as well as re-enabling the pin interrupts when the fucntion exits.